### DISCUSSION TITLE
Lab Meeting Feb 13, 2023: Tidy Tuesday

### DISCUSSION THREAD

**Discussion by AHuffmyer** [RobertsLab/resources#1574] [General]
For lab meeting on Feb 13, we are going to participate in a group Tidy Tuesday exercise! Follow the steps below. 

Step 1: Visit the TidyTuesday website and check out the resources and podcast episodes: https://www.tidytuesday.com/

Step 2: Go to the TIdyTuesday GitHub repo and read the README: https://github.com/rfordatascience/tidytuesday

Step 3: Go to the 2023 bird watching dataset and read the README and the data dictionary information: https://github.com/rfordatascience/tidytuesday/blob/master/data/2023/2023-01-10/readme.md

Step 4: Download the data set and subset using this code in R. Note that Option 1 worked for me and was the easiest. Options 2 and 3 are other ways to get the data, but if Option 1 worked you are all set!  

```
library(tidyverse)

# Option 1: Read in manually

feederwatch <- readr::read_csv('https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2023/2023-01-10/PFW_2021_public.csv')
site_data <- readr::read_csv('https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2023/2023-01-10/PFW_count_site_data_public_2021.csv')


# Option 2: download with TidyTuesday package

# Read in with tidytuesdayR package 
# Install from CRAN via: install.packages("tidytuesdayR")
# This loads the readme and all the datasets for the week of interest

# Either ISO-8601 date or year/week works!

tuesdata <- tidytuesdayR::tt_load('2023-01-10')
tuesdata <- tidytuesdayR::tt_load(2023, week = 02)

feederwatch <- tuesdata$feederwatch


# Option 3: Download the raw data and subset 

PFW_2021_public <- readr::read_csv("https://clo-pfw-prod.s3.us-west-2.amazonaws.com/data/PFW_2021_public.csv")
dplyr::glimpse(PFW_2021_public)

# There are almost three million rows! The file is too big for github, let's
# subsample.

set.seed(424242)
PFW_2021_public_subset <- dplyr::slice_sample(PFW_2021_public, n = 1e5)

readr::write_csv(PFW_2021_public_subset, here::here("data", "2023", "2023-01-10", "PFW_2021_public.csv"))

```

Step 5: Now that you know what types of data are available, write down 1-2 questions you want to answer using this dataset (e.g., how does the number of birds sited relate to the habitat type?) 

Step 6: Manipulate the data (make it tidy!) and make some fun plots! Post your favorite plot to this discussion thread. In lab meeting we will go around the room and talk about what questions we had, how we manipulated the data and any issues we ran into, and the plots we made. 

Step 7: Keep up with Tidy Tuesday in the future if you want! 

The goal for the meeting is to make a plot, no matter how simple. Don't take too much time for this exercise. As you go through the exercise, think about areas of coding that you'd like to learn more about or get better at that we can plan for a future meeting.

Let Ariana know if you have any questions.

**Comment by kubu4** [comment]
Things are breaking at this step:

```r
> set.seed(424242)
> PFW_2021_public_subset <- dplyr::slice_sample(PFW_2021_public, n = 1e5)
> 
> readr::write_csv(PFW_2021_public_subset, here::here("data", "2023", "2023-01-10", "PFW_2021_public.csv"))
Warning in open.connection(file, "wb") :
  cannot open file '/home/shared/8TB_HDD_01/sam/gitrepos/RobertsLab/tidy-tues-20230110/data/2023/2023-01-10/PFW_2021_public.csv': No such file or directory
Error in open.connection(file, "wb") : cannot open the connection
```

Do I need to create those subdirectories and filename _before_ running that `readr::write_csv()` command?

Also, I noticed that the `feederwatch` variable contents are listed as `NULL`. Is that expected?

**Reply by AHuffmyer** [reply]
I have revised the instructions - use Option 1 above to read the data in manually. The code is below: 

```
library(tidyverse)

# Option 1: Read in manually

feederwatch <- readr::read_csv('https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2023/2023-01-10/PFW_2021_public.csv')
site_data <- readr::read_csv('https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2023/2023-01-10/PFW_count_site_data_public_2021.csv')
```

**Reply by kubu4** [reply]
Thanks, that worked!

**Comment by kubu4** [comment]
Probably should have mentioned the `feederwatch` issue separately. Anyway, when I run

```r
feederwatch <- tuesdata$feederwatch
```

It results in a `NULL` value for `feederwatch`. Looking at the data, I guess this makes sense, as there's nothing in `tuesdata` object which matches `$feederwatch` (at least, based on autocompletion).

Any idea why this aspect of the code is present if it doesn't actually load anything into the `feederwatch` variable?

**Reply by AHuffmyer** [reply]
Thanks for bringing up the issues! I'll run through it and post the solutions/revised code.

**Reply by AHuffmyer** [reply]
If you read data in manually using Option 1 (above) you shouldn't have an issue and wont need this step :)

**Reply by kubu4** [reply]
Thanks, that worked!

**Comment by kubu4** [comment]
Alrighty, I'm going to admit that with my limited experience visualizing data, I've found this task is a pretty big challenge and has eaten up a fair amount of time (with little to show for it). I'd definitely vote for something more like a guided "how-to"!

For others that might also be struggling (hopefully I'm not the only one?!?!), here are some helpful tips that I've managed to use:

1. Use the `tidyverse` and `ggplot2` packages (this is part of "tidy" Tuesday, after all):

    ```r
    # Load libraries
    library("tidyverse")
    library("ggplot2")
    ```

2. Try to figure out some stuff using the `group_by()` and `summarise()` functions. Here's an extremely limited example of something I managed:

    ```r
    # Look at species counts by Month
    feederwatch %>%
    group_by(Month, species_code) %>%
    summarise(how.many = sum(how_many))


    # BELOW HERE IS THE OUTPUT FROM THE ABOVE COMMAND
    
    `summarise()` has grouped output by 'Month'. You can override using the `.groups` argument.
    # A tibble: 1,373 x 3
    # Groups:   Month [6]
    Month species_code how.many
    <dbl> <chr>           <dbl>
    1     1 abetow             20
    2     1 accipi              3
    3     1 acowoo             28
    4     1 allhum             25
    5     1 amecro           1030
    6     1 amegfi           5712
    7     1 amekes              2
    8     1 amerob           1029
    9     1 amtspa            499
   10     1 anhing              1
   # … with 1,363 more rows
    ```

3. Then, I've only been able to figure out how to plot a single species' counts, within a specific geographic region, across months, so here's that. Also, I've managed to do so with the added bonus of changing the interpretation of Month numbers so that they aren't interpreted as actual numbers and are instead interpreted as characters. This makes it so the X-axis doesn't attempt to scale as a continuous set of numbers. Also addd the `xlab()` function to change the X-axis label to "Month". Haven't figured out how to rearrange the month numbers to be in sorted in normal human numerical sorting...:

    ```r
    # Dot plot showing only "amegfi" species counts within "US-IL" region in each month
    # Use as.character() function on the "Month" column to prevent X-axis scaling based on numbers
    feederwatch %>%
    filter(species_code == "amegfi", subnational1_code == "US-IL") %>%
    ggplot(aes(x = as.character(Month), y = how_many)) +
        geom_point() +
        xlab("Month")
    ```

And, here's that plot:

![Dot plot showing only "amegfi" species counts within "US-IL" region in each month ](https://user-images.githubusercontent.com/4514104/217937234-118deafa-3d44-4cfa-8eb1-23c14418578b.png)


---

Getting to this point took a long time. Anyway, hopefully sharing this helps people get an idea of how/where to start!

Looking forward to lab meeting to see some fancy stuff and to "steal" some code for better understanding/plots!

**Comment by AHuffmyer** [comment]
Hey Sam! This looks great! You did all you needed to do - make a plot! It definitely doesn't need to take up too much of anybody's time, so just do what you are able/want to do beforehand. We will talk through all the points you raised at the meeting. This will be a good exercise to identify areas where we would like to learn more coding skills - like maybe a tidyverse tutorial at a future meeting!

**Comment by AHuffmyer** [comment]
Here is a notebook post with code and plots that I made!

https://ahuffmyer.github.io/ASH_Putnam_Lab_Notebook/TidyTuesday-Exercises/
